policyengine_us/variables/gov/states/mt/tax/income/mt_income_tax_before_non_refundable_credits_indiv.py

```diff
@@ -10,24 +10,44 @@ class mt_income_tax_before_non_refundable_credits_indiv(Variable):
     defined_for = StateCode.MT
 
     def formula(person, period, parameters):
-        income = person("mt_taxable_income_indiv", period)
         p = parameters(period).gov.states.mt.tax.income.main
+        regular_income = person("mt_taxable_income_indiv", period)
         filing_status = person.tax_unit(
             "state_filing_status_if_married_filing_separately_on_same_return",
             period,
         )
         status = filing_status.possible_values
-        return select(
-            [
-                filing_status == status.SINGLE,
-                filing_status == status.HEAD_OF_HOUSEHOLD,
-                filing_status == status.SEPARATE,
-                filing_status == status.SURVIVING_SPOUSE,
-            ],
-            [
-                p.single.calc(income),
-                p.head_of_household.calc(income),
-                p.separate.calc(income),
-                p.widow.calc(income),
-            ],
-        )
+        if p.capital_gains.availability:
+            capital_gains = person("long_term_capital_gains", period)
+            taxable_income = max_(regular_income - capital_gains, 0)
+            capital_gains_tax = person("mt_capital_gains_tax_indiv", period)
+            regular_tax = select(
+                [
+                    filing_status == status.SINGLE,
+                    filing_status == status.HEAD_OF_HOUSEHOLD,
+                    filing_status == status.SEPARATE,
+                    filing_status == status.SURVIVING_SPOUSE,
+                ],
+                [
+                    p.single.calc(taxable_income),
+                    p.head_of_household.calc(taxable_income),
+                    p.separate.calc(taxable_income),
+                    p.widow.calc(taxable_income),
+                ],
+            )
+            return regular_tax + capital_gains_tax
+        else:
+            return select(
+                [
+                    filing_status == status.SINGLE,
+                    filing_status == status.HEAD_OF_HOUSEHOLD,
+                    filing_status == status.SEPARATE,
+                    filing_status == status.SURVIVING_SPOUSE,
+                ],
+                [
+                    p.single.calc(regular_income),
+                    p.head_of_household.calc(regular_income),
+                    p.separate.calc(regular_income),
+                    p.widow.calc(regular_income),
+                ],
+            )
```

Comment:
```suggestion
        regular_tax = select(
                [
                    filing_status == status.SINGLE,
                    filing_status == status.HEAD_OF_HOUSEHOLD,
                    filing_status == status.SEPARATE,
                    filing_status == status.SURVIVING_SPOUSE,
                ],
                [
                    p.single.calc(taxable_income),
                    p.head_of_household.calc(taxable_income),
                    p.separate.calc(taxable_income),
                    p.widow.calc(taxable_income),
                ],
            )
        if p.capital_gains.availability:
            capital_gains = person("long_term_capital_gains", period)
            taxable_income = max_(regular_income - capital_gains, 0)
            capital_gains_tax = person("mt_capital_gains_tax_indiv", period)
            return regular_tax + capital_gains_tax
        else:
            return regular_tax
```

Reply:
same in _joint

Reply:
If it does then feel free to change variable names as needed, the concept is to avoid duplicating code in the if and else sides

Reply:
Got it : )